lib/client/api_matrix.es6.js

```diff
@@ -0,0 +1,129 @@
+foam.CLASS({
+  name: 'ApiMatrix',
+  package: 'org.chromium.apis.web',
+  documentation: `ApiMatrix is a client side object that has methods
+    to retrieve browserAPIs from server DAO, transform browserAPIs to
+    CSV format string, tranform browserAPIs to object that can be
+    displayed on HTML as a nested table.`,
+  requires: [
+    'foam.dao.EasyDAO',
+    'foam.dao.ArrayDAO',
+    'foam.mlang.ExpressionsSingleton',
+    'org.chromium.apis.web.BrowserAPI',
+  ],
+  properties: [
+    {
+      name: 'browserAPIs',
+      documentation: `The WebAPIs object where the browserAPIs will be
+        be fetched from.`,
+      typeName: 'BrowserAPI DAO',
+      required: true,
+      final: true,
+    },
+    {
+      name: 'M',
+      documentation: `Object contains mlang expressions, it supports
+        logic operations such as AND and OR and math operations such
+        as IN, EQ.`,
+      factory: function() {
+        return this.ExpressionsSingleton.create();
+      },
+    },
+  ],
+  methods: [
+    {
+      name: 'toMatrix',
+      documentation: `An asynchronous method that takes an array of browser
+        keys, produce a nested matrix that contains browser and interface
+        relationship. This structure should be easy to be displayed as a
+        nested table in client side.`,
+      args: [
+        {
+          name: 'browserKeys',
+          typeName: 'StringArray',
+          documentation: `A string array contains valid browser keys. It will
+            be used for filtering browserAPIs.`,
+        },
+      ],
+      returns: {
+        typeName: 'JSON',
+        documentation: `Returns a JSON of the form:
+            {interfaceName: {apiName: {BrowserKey: true, ...} ...} ... } ...}
+            This structure is easy for displaying a nested table.`,
+      },
+      code: function(browserKeys) {
+        return this.browserAPIs
+          .where(this.M.IN(this.BrowserAPI.BROWSER_KEY, browserKeys))
+          .select(this.M.GROUP_BY( this.BrowserAPI.INTERFACE_KEY,
+            this.ArrayDAO.create()))
```

Comment:
Changed to ArraySInk, I think both of them works though. ArrayDAO.array and ArraySink.a contains the same result..